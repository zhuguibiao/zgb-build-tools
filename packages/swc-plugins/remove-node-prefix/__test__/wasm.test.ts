import { expect, test } from "vitest";
import { transform } from "@swc/core";
import path from "node:path";
import url from "node:url";
import fs from "node:fs/promises";

const pluginName = "swc_plugin_remove_node_prefix.wasm";
const transformCode = async (code: string, options = {}) => {
  return transform(code, {
    jsc: {
      parser: {
        syntax: "ecmascript",
      },
      target: "es2018",
      experimental: {
        plugins: [
          [
            path.join(
              path.dirname(url.fileURLToPath(import.meta.url)),
              "..",
              pluginName,
            ),
            options,
          ],
        ],
      },
    },
    filename: "test.js",
  });
};

test("Should load remove-console wasm plugin correctly", async () => {
  const input = await fs.readFile(
    new URL("./fixtures/input.js", import.meta.url),
    "utf-8",
  );

  const { code } = await transformCode(input);
  console.log('input:---',input)
  console.log('output:---',code)
  expect(code).toMatchSnapshot();
});
