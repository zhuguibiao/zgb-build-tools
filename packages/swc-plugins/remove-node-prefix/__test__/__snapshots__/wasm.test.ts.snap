// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should load remove-console wasm plugin correctly 1`] = `
"import fs from 'fs';
import path from 'path';
const dynamic = import('url');
const fs2 = require('fs');
const path2 = require.resolve('path');
import * as _fs3 from 'fs';
export { _fs3 as fs3 };
export * from 'crypto';
export { readFile } from 'fs';
module.exports = {
    fs: require('fs')
};
module.exports.fs = require('fs');
/** ignore */ console.log('node:fs');
fn('node:fs');
"
`;
